### PR TITLE
[serve] raise RequestCancelledError when request is cancelled during assignment

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

Currently `ray.serve.exceptions.RequestCancelledError` is raised only when a request is cancelled during execution. We should also raise `RequestCancelledError` when a request is cancelled during assignment.

### CODE REVIEW

**Review by GeneDer — APPROVED** [review]
:shipit:

File: python/ray/serve/handle.py [hunk 1]
```diff
@@ -344,9 +346,12 @@ async def _fetch_future_result_async(self) -> ReplicaResult:
...
             # safely from any asyncio loop.
-            self._replica_result = await asyncio.wrap_future(
-                self._replica_result_future
-            )
+            try:
+                self._replica_result = await asyncio.wrap_future(
+                    self._replica_result_future
+                )
+            except asyncio.CancelledError:
+                raise RequestCancelledError from None
```
Comment (edoakes) [inline python/ray/serve/handle.py:354]: Doesn't this take the request ID as an arg?

Comment (zcin) [reply python/ray/serve/handle.py:354]: hmmm yeah, it's optional though. let me see if we can get ahold of the request metadata here.

Comment (zcin) [reply python/ray/serve/handle.py]: added request ID, ptal!

Comment (edoakes) [reply python/ray/serve/handle.py]: maybe let's make it required so this is consistent

**Review by edoakes — APPROVED** [review]
nice!

**Pull request merged**